services/task-svc/internal/patient/patient.go

```diff
@@ -472,6 +472,29 @@ func (ServiceServer) GetPatientDetails(ctx context.Context, req *pb.GetPatientDe
 	}, nil
 }
 
+func (ServiceServer) mapTasks(tasks []models.Task) []*pb.GetPatientListResponse_Task {
```

Comment:
```suggestion
func mapTasks(tasks []models.Task) []*pb.GetPatientListResponse_Task {
```
Good for reusability, but this function should not be a member of `ServiceServer`. You could also put the function inside `GetPatientList` like `mapTasks := func () {}`.

Reply:
Jup that was excatly the thing I was struggling with, thank you!

Reply:
@MaxSchaefer should be fixed now.